### PR TITLE
Various fixes and improvements. Use latest Security Group module. Bump module versions. Update GitHub workflows

### LINKED ISSUE #77: LoadBalancer coming blank in beanstalk UI
screen shot 2019-02-11 at 12 01 37 pm

LoadBalancer coming blank in ElasticBeanstalk UI. Cannot see/list/edit options such as instanceport, https, routing, ssl certificates s3 log enabled etc. attached is the screenshot.

### LINKED ISSUE #78: when beanstalk setting is blank
for example like SSLPolicy

the error returns:
Creating Load Balancer listener failed Reason: An SSL policy must be specified for HTTPS listeners

### LINKED ISSUE #89: allow to use custom AMI Imageid
this is an feature suggestion

it would be nice if the module would expose an option to customize the AMI ImageId used by beanstalk:

```
setting {
    name      = "ImageId"
    namespace = "aws:autoscaling:launchconfiguration"
    value     = var.custom_ami_imageid
  }
```

### LINKED ISSUE #118: Error: ConfigurationValidationException on a apply
Found a bug? Maybe our [Slack Community](https://slack.cloudposse.com) can help. 

[Slack Community](https://slack.cloudposse.com)

## Describe the Bug
Plan goes ok.
After trying to apply my config, a **ConfigurationValidationException** pops up

## Expected Behavior
Clearer meaning of the problem. I think I used all the required vars documented... so no clue how to proceed

## Steps to Reproduce
`
module "elastic_beanstalk_application" {
  source      = "git::https://github.com/cloudposse/terraform-aws-elastic-beanstalk-application.git?ref=tags/0.5.0"
  name        = "AppName"
}

module "elastic-beanstalk-environment" {
  source  = "cloudposse/elastic-beanstalk-environment/aws"
  version = "0.22.0"
  elastic_beanstalk_application_name = "AppName"
  name = "dev-AppName"
  vpc_id = aws_default_vpc.default.id
  region = "eu-west-1"
  solution_stack_name = "64bit Amazon Linux 2 v5.0.2 running Node.js 12"
  application_subnets = [
    aws_default_subnet.default_az1.id,
    aws_default_subnet.default_az2.id,
    aws_default_subnet.default_az3.id,
  ]
  autoscale_max = 2
  env_vars = {
    API_URL = "https://my-api-url.net"
  }
}
`

## Screenshots
Captura de Pantalla 2020-06-04 a les 13 15 48
`
Error: ConfigurationValidationException: Configuration validation exception: Invalid option value: '' (Namespace: 'aws:ec2:vpc', OptionName: 'ELBSubnets'): Specify the subnets for the VPC.
	status code: 400, request id: ac1268fc-77a0-479d-8a5d-37c6b024ce9b

  on .terraform/modules/elastic-beanstalk-environment/terraform-aws-elastic-beanstalk-environment-0.22.0/main.tf line 505, in resource "aws_elastic_beanstalk_environment" "default":
 505: resource "aws_elastic_beanstalk_environment" "default" {
`

## Environment:
 - OS: OSX
 - Version Dockerized Terraform v0.12.21

### LINKED ISSUE #145: Wrong Descriptions of additional_security_groups  and allowed_security_groups
##  Describe the Bug
In the README, under the **Inputs** table, the descriptions of `additional_security_groups` and `allowed_security_groups` input fields are switched. This is merely an issue in the README documentation - the module behaves according to the *expected* descriptions below.

## Current

The description in the README currently shows

additional_security_groups: List of security groups to be allowed to connect to the EC2 instances
allowed_security_groups: List of security groups to add to the EC2 instances

## Expected

additional_security_groups: List of security groups to add to the EC2 instances
allowed_security_groups: List of security groups to be allowed to connect to the EC2 instances

### PR DESCRIPTION
**Pull request by aknysh**
## what
* Various fixes and improvements
* Use latest Security Group module
* Bump module versions
* Update GitHub workflows
* Update/fix tests
* Add example and test for Elastic Beanstalk with a shared ALB
* Add `enabled` logic to all resources and data sources

## why
* Standardize on using https://github.com/cloudposse/terraform-aws-security-group in all modules
* Keep up to date
* Cleanup

## related
* Closes #202 
* Closes #190 
* Closes #201 
* Closes #199 
* Closes #198 
* Closes #197 
* Closes #196 
* Closes #195 
* Closes #193 
* Closes #189 
* Closes #188 
* Closes #184 
* Closes #183 
* Closes #145 
* Closes #118 
* Closes #89 
* Closes #78 
* Closes #77

### CODE REVIEW

File: versions.tf [hunk 1]
```diff
@@ -1,10 +1,10 @@
 terraform {
-  required_version = ">= 0.13.0"
+  required_version = ">= 0.14.0"
 
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = ">= 2.0"
+      version = ">= 3.0"
```
Comment (florian0410) [inline versions.tf:7]: Is there a particular reason for this to be version 3.0 but not in the examples ?

**Comment by florian0410** [comment]
For my original PR #201, I don't see any issue. So it is good for me. 

Thank you for the huge refresh !

File: README.yaml [hunk 2]
```diff
@@ -76,108 +79,117 @@ usage: |-
...
     }
-
+    
     module "vpc" {
-      source     = "git::https://github.com/cloudposse/terraform-aws-vpc.git?ref=tags/0.8.0"
-      namespace  = var.namespace
-      stage      = var.stage
-      name       = var.name
+      source     = "cloudposse/vpc/aws"
+      version    = "0.28.1"
```
Comment (korenyoni) [inline README.yaml:85]: This is what we do in other module README example snippets: https://github.com/cloudposse/terraform-aws-vpc#examples

```suggestion
      # Cloud Posse recommends pinning every module to a specific version
      # version = "x.x.x"
```

File: README.yaml [hunk 3]
```diff
@@ -76,108 +79,117 @@ usage: |-
...
       cidr_block = "172.16.0.0/16"
+  
+      context = module.this.context
     }
-
+    
     module "subnets" {
-      source               = "git::https://github.com/cloudposse/terraform-aws-dynamic-subnets.git?ref=tags/0.16.0"
+      source               = "cloudposse/dynamic-subnets/aws"
+      version              = "0.39.8"
```
Comment (korenyoni) [inline README.yaml:94]: ditto

```suggestion
      # Cloud Posse recommends pinning every module to a specific version
      # version = "x.x.x"
```

File: README.yaml [hunk 4]
```diff
@@ -76,108 +79,117 @@ usage: |-
...
-
+    
     module "elastic_beanstalk_application" {
-      source      = "git::https://github.com/cloudposse/terraform-aws-elastic-beanstalk-application.git?ref=tags/0.3.0"
-      namespace   = var.namespace
-      stage       = var.stage
-      name        = var.name
-      description = "Test elastic_beanstalk_application"
+      source      = "cloudposse/elastic-beanstalk-application/aws"
+      version     = "0.11.1"
```
Comment (korenyoni) [inline README.yaml:108]: ditto

```suggestion
      # Cloud Posse recommends pinning every module to a specific version
      # version = "x.x.x"
```

File: examples/complete/versions.tf [hunk 5]
```diff
@@ -1,10 +1,10 @@
...
-  required_version = ">= 0.13.0"
+  required_version = ">= 0.14.0"
 
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = ">= 2.0"
+      version = ">= 3.0"
     }
     null = {
```
Comment (korenyoni) [inline examples/complete/versions.tf:9]: We're not using this provider

```bash
module (terraform_unused_required_providers)

  on versions.tf line 9:
   9:     null = {
  10:       source  = "hashicorp/null"
  11:       version = ">= 2.0"
  12:     }

Reference: https://github.com/terraform-linters/tflint/blob/v0.27.0/docs/rules/terraform_unused_required_providers.md
```

File: examples/shared-alb/versions.tf [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 2.0"
+    }
+    null = {
+      source  = "hashicorp/null"
+      version = ">= 2.0"
+    }
```
Comment (korenyoni) [inline examples/shared-alb/versions.tf:12]: We're not using this provider (see TFLint output in other comments in my review)

```suggestion
```

File: main.tf [hunk 7]
```diff
@@ -1,3 +1,9 @@
+data "aws_partition" "current" {}
```
Comment (korenyoni) [inline main.tf:1]:
```suggestion
data "aws_partition" "current" {
  count = local.enabled ? 1 : 0
}
```

File: main.tf [hunk 8]
```diff
@@ -1,3 +1,9 @@
...
+
+locals {
```
Comment (korenyoni) [inline main.tf:3]:
```suggestion
locals {
  enabled = module.this.enabled
```

File: main.tf [hunk 9]
```diff
@@ -25,12 +31,12 @@ resource "aws_iam_role" "service" {
 resource "aws_iam_role_policy_attachment" "enhanced_health" {
   count      = var.enhanced_reporting_enabled ? 1 : 0
```
Comment (korenyoni) [inline main.tf:32]:
```suggestion
  count      = local.enabled && var.enhanced_reporting_enabled ? 1 : 0
```

File: main.tf [hunk 10]
```diff
@@ -1,3 +1,9 @@
+data "aws_partition" "current" {}
+
+locals {
+  partition = data.aws_partition.current.partition
```
Comment (korenyoni) [inline main.tf:4]:
```suggestion
  partition = join("", data.aws_partition.current.*.partition)
```

**Review by korenyoni — COMMENTED** [review]
Watch out for cases where `module.this.enabled=false`, also clean up `versions.tf` in accordance to TFLint in examples

File: versions.tf (hunk 1 above)
Comment (korenyoni) [reply versions.tf:7]: Yeah we should update the examples to have a consistent `versions.tf`

Comment (aknysh) [reply versions.tf:7]: updated

**Review by korenyoni — APPROVED** [review]
LGTM.

**Pull request merged**